py4web/core.py

```diff
@@ -1138,65 +1141,87 @@ def keyboardInterruptHandler(signal, frame):
 @click.argument('apps_folder', default='apps')
-@click.option('-Y', '--yes', is_flag=True, default=False, help='No prompt, assume yes to questions')
+@click.option('-Y', '--yes', is_flag=True, default=False, help='No prompt, assume yes to questions', show_default=True)
 def setup(**args):
+    """Setup new apps folder or reinstall it"""
     install_args(args, reinstall_apps=True)
 
 
 @cli.command()
-@click.argument('apps_folder', default='apps')
+@click.argument('apps_folder', default='apps', type=click.Path(exists=True))
 def shell(apps_folder):
+    """Open a python shell with apps_folder added to the path"""
     install_args(dict(apps_folder=apps_folder))
     fix_ansi_on_windows()
     code.interact(local=dict(globals(), **locals()))
 
 
 @cli.command()
-@click.argument('apps_folder')
-@click.argument('func')
-@click.option('--args', default='{}')
-def call(apps_folder, func, args):
+@click.argument('apps_folder', type=click.Path(exists=True))
+@click.argument('function')
+@click.option('--args', default='{}', help = 'Arguments passed to the program/function', show_default=True)
+def call(apps_folder, function, args):
+    """Call a function inside apps_folder"""
     args = json.loads(args)
     install_args(dict(apps_folder=apps_folder))
-    module, name = ('apps.' + func).rsplit('.', 1)
+    module, name = ('apps.' + function).rsplit('.', 1)
     env = {}
     if not apps_folder in sys.path:
         sys.path.insert(0, apps_folder)
     exec('from %s import %s' % (module, name), {}, env)
     env[name](**args)
 
 @cli.command()
-@click.option('--password', prompt=True, confirmation_prompt=True, hide_input=True)
-@click.option('-p', '--password_file', default='password.txt', help='File for the encrypted password')
+@click.option('--password', prompt=True, confirmation_prompt=True, hide_input=True, help = 'Password value (asked if missing)')
+@click.option('-p', '--password_file', default='password.txt', help='File for the encrypted password', show_default=True)
 def set_password(password, password_file):
+    """Set administrator's password for the Dashboard"""
     click.echo('Storing the hashed password in file "%s"\n' % password_file)
     with open(password_file, "w") as fp:
         fp.write(str(pydal.validators.CRYPT()(password)[0]))
 
 @cli.command()
-@click.argument('apps_folder', default='apps')
-@click.option('-Y', '--yes', is_flag=True, default=False, help='No prompt, assume yes to questions')
-@click.option('-H', '--host', default='127.0.0.1', help='Host name')
-@click.option('-P', '--port', default=8000, type=int, help='Port number')
-@click.option('-p', '--password_file', default='password.txt', help='File for the encrypted password')
-@click.option('-w', '--number_workers', default=0, type=int, help='Number of workers')
-@click.option('-d', '--dashboard_mode',  default='full', help='Dashboard mode: demo, readonly, full (default), none')
+@click.argument('apps_folder', default='apps', type=click.Path(exists=True))
```

Comment:
@nicozanf you're welcome!
But why do you require apps-folder to exist? (for run-command at least)
If there is no apps-folder, it is created during `intstall_args`
I think this is causing the test to fail     
https://github.com/web2py/py4web/blob/e92dc89c841e958c822ab1227476b06b8e9bb537/py4web/core.py#L1070-L1085

Reply:
I misses it, thank you again @valq7711 !